pkg/api/customization/globalnamespaceaccess/access_common.go

```diff
@@ -126,50 +125,170 @@ func (ma *MemberAccess) EnsureRoleInTargets(targetProjects, roleTemplates []stri
 		}
 		clusterName := split[0]
 		projectName := split[1]
-
-		if !projects[projectName] {
-			projects[projectName] = true
-		}
+		projectToClusterMap[projectName] = clusterName
```

Comment:
I hope this projectName is actually Id? Since there can be two projects with same name in two clusters, making sure what we have here is unique key

Reply:
Right! projectIds are generated with random chars but there's a chance they could end up being the same in some clusters. I'll stick to clusterId:projectId